In [ ]:
#Cargamos el mismo dataset de la google store
import numpy as np
import pandas as pd
import locale

url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

In [3]:
#RETO 1 -> La trampa de las unidades de medida

def convertToNum(numStr : str) -> float:

    if numStr.endswith("M"): #Represent megabytes
        numStr = numStr[:-1]
        return float(numStr)
    elif numStr.endswith("k"):
        numStr = numStr[:-1]
        newVal = float(numStr) / 1024 #Convert to megabytes
        return newVal

    return np.nan

df['Size'] = df['Size'].apply(convertToNum)
print(df['Size'].mean())


21.51616543577433


 Una vez convertida la columna a valores numéricos (Megabytes), ejecuta el método .mean(). ¿Cuál es el peso promedio en Megabytes de las apps en la Play Store? El promedio es de 21.516 megabytes 

In [ ]:
#RETO 2 -> El Tipo de Dato Cronológico
locale.setlocale(locale.LC_TIME, 'en_US.UTF-8') 
df['Last Updated'] = pd.to_datetime(df['Last Updated'], format='%B %d, %Y') #Convert to specific datetime using this format
print(df['Last Updated'].head(20))


0    NaT
1    NaT
2    NaT
3    NaT
4    NaT
5    NaT
6    NaT
7    NaT
8    NaT
9    NaT
10   NaT
11   NaT
12   NaT
13   NaT
14   NaT
15   NaT
16   NaT
17   NaT
18   NaT
19   NaT
Name: Last Updated, dtype: datetime64[ns]
